### PR TITLE
added auto tagging to the google ads final url performance report

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

This PR introduces the functionality to enable auto tagging for UTM fields within the Google Ads FinalURLPerformance Report.

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [X] No  (please provide explanation how the change is non breaking below.)

This PR only introduces a conditional to handle a coalesce for users that are leveraging the auto tagging feature. 

**Is this PR in response to a previously created Issue**

- [X] Yes, Issue #15 
- [ ] No 

**How did you test the PR changes?** 

 
- [X] CircleCi  
- [ ] Other (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 
- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🥼 
**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,6 +101,12 @@ vars:
     google_ads_schema: your_schema_name
     google_ads_database: your_database_name
 ```
+### UTM Auto Tagging Feature
+This package assumes you are manually adding UTM tags to the `EffectiveFinalUrl` field within the `FINAL_URL_REPORT`. If you are leveraging the auto-tag feature within Google Ads then you will want to enable the `google_auto_tagging_enabled` variable to correctly populate the UTM fields within the `stg_google_ads__final_url_performance` model.
```
Comment (fivetran-joemarkiewicz) [inline README.md]:
```suggestion
This package assumes you are manually adding UTM tags to the `EffectiveFinalUrl` field within the `FINAL_URL_REPORT` table. If you are leveraging the auto-tag feature within Google Ads then you will want to enable the `google_auto_tagging_enabled` variable to correctly populate the UTM fields within the `stg_google_ads__final_url_performance` model.
```

File: models/stg_google_ads__final_url_performance.sql [hunk 2]
```diff
@@ -27,11 +27,25 @@ url_fields as (
...
         {{ dbt_utils.get_url_host('final_url') }} as url_host,
         '/' || {{ dbt_utils.get_url_path('final_url') }} as url_path,
+
+        {% if var('google_auto_tagging_enabled', false) %}
+
+        coalesce( {{ dbt_utils.get_url_parameter('final_url', 'utm_source') }} , 'google')  as utm_source,
+        coalesce( {{ dbt_utils.get_url_parameter('final_url', 'utm_medium') }} , 'cpc') as utm_medium,
+        coalesce( {{ dbt_utils.get_url_parameter('final_url', 'utm_campaign') }} , campaign_name) as utm_campaign,
+        coalesce( {{ dbt_utils.get_url_parameter('final_url', 'utm_content') }} , ad_group_name) as utm_content,
+        {{ dbt_utils.get_url_parameter('final_url', 'utm_term') }} as utm_term
```
Comment (fivetran-jamie) [inline models/stg_google_ads__final_url_performance.sql]: perhaps this line can live outside of the if/else blocks

**Review by fivetran-jamie — APPROVED** [review]
lgtm! one stylistic suggestion 😄

File: README.md [hunk 3]
```diff
@@ -101,6 +101,12 @@ vars:
     google_ads_schema: your_schema_name
     google_ads_database: your_database_name
 ```
+### UTM Auto Tagging Feature
+This package assumes you are manually adding UTM tags to the `EffectiveFinalUrl` field within the `FINAL_URL_REPORT` table. If you are leveraging the auto-tag feature within Google Ads then you will want to enable the `google_auto_tagging_enabled` variable to correctly populate the UTM fields within the `stg_google_ads__final_url_performance` model.
+```yml
+vars:
+    google_auto_tagging_enabled: true # False by default
```
Comment (fivetran-jamie) [inline README.md]: should we include package-scope here?

**Pull request merged**